test-utils-base/src/commonMain/kotlin/mock/MockMethod.kt

```diff
@@ -150,24 +157,59 @@ sealed class BaseMethodMock<
      * Resets all calls to this mock method. This means that `verify(times=0)` should succeed
      */
     fun resetCalls() {
-        callParameters.clear()
+        callParametersStateFlow.value = listOf()
     }
 
     /**
-     * Removes all stubbs from this mock method. Note that this also removes any default stubbs that may have been created when declaring the mock.
+     * Removes all stubs from this mock method. Note that this also removes any default stubs that may have been created when declaring the mock.
      */
     fun resetStubs() {
         stubs.clear()
     }
 
     /**
-     * Resets both calls and stubbs. Shorthand for [resetCalls] and [resetStubs]
+     * Resets both calls and stubs. Shorthand for [resetCalls] and [resetStubs]
      */
     fun reset() {
         resetCalls()
         resetStubs()
     }
 
+    internal suspend fun verifyWithParametersWithin(duration: Duration, parameters: C, times: Int = 1) {
+        verifyWithParametersWithin(duration, parameters, VerificationRule.times(times))
+    }
+
+    internal suspend fun verifyWithParametersWithin(duration: Duration, parameters: C, verificationRule: VerificationRule) = parametersSpec.apply {
```

Comment:
This method is what I've been dreaming about for a long time!

Reply:
`yieldMultiple` 🚮